Python pyproject package

TheStaticTurtle opened the pull request:
Hi,

I've taken some time to adapt the python binding in order to be able to build wheels.

This includes:
 - Writing the `pyproject.toml`
 - Modifying the build script to manually link boost and adding the install statement to include the pyd into the wheel

Additionally I added a `FetchContent` for boost 1.89 int the `deps` cmake file so that the build process "just works". I have not tested if this causes an issue somewhere else!

I've successfully used `pip wheel .` to build the project with python 3.13 on windows.

jcelerier commented on pyproject.toml:
```diff
@@ -0,0 +1,29 @@
+    { name = "Celerier Jean-Michaël", email = "jeanmichael.celerier@gmail.com" },
+]
+classifiers = [
+    "License :: BSD2",
```
Is there a way to identify the licenses of the project? They aren't a "OR" but a "AND"

TheStaticTurtle replied:
My bad, didn't read enough of the readme.

According to the packaging guide 
> As per [PEP 639](https://peps.python.org/pep-0639/) licenses should be declared with two fields:
> - `license` is an [SPDX license expression](https://packaging.python.org/en/latest/glossary/#term-License-Expression) consisting of one or more [license identifiers](https://packaging.python.org/en/latest/glossary/#term-License-Identifier).
> - `license-files` is a list of license file glob patterns.

So yes I can specify multiple licenses but I don't think I can specify where they apply. would the following be ok?:
```toml
license-files = [
    "LICENSE.md"
]
license = "MIT AND BSD-2-Clause"
```

TheStaticTurtle commented:
Hi,

As requested I updated the license and authors.

I also added an explicit check for `readerwriterqueue` in the cmake of the binding because it sometimes refused to build the first time for some reason and I don't know enough to figure out why.

> Is there a reason this requires scikit ?

Maybe not but it's the easiest/fastest way I found to integrate cmake+nanobind in the packaging process.

> Do you know if this builds with uv too?

It does.

There is an edge case where if it needs to download boost, it might fail due to the file path/name being too long. This can be fixed by tweaking the cache settings of uv.
